File: src/test/scala/com/cenerino/algorithms/scala/misc/ClockHandsAngleTest.scala
package com.cenerino.algorithms.scala.misc

import com.cenerino.algorithms.scala.misc.ClockHandsAngle._
import org.scalatest.FunSuite

class ClockHandsAngleTest extends FunSuite {

  test("calculate angle @ 00:00") {
    calcAngle(0, 0) === 0
  }

  test("calculate angle @ 00:30") {
    calcAngle(0, 30) === 180
  }

  test("calculate angle @ 03:45") {
    calcAngle(3, 45) === 180
  }

  test("calculate angle @ 12:45") {
    calcAngle(12, 45) === 270
  }

  test("calculate angle @ 15:00") {
    calcAngle(15, 0) === 90
  }

  test("calculate angle @ 23:55") {
    calcAngle(23, 55) === 0
  }

  test("calculate angle @ 23:59") {
    calcAngle(23, 59) === 24
  }

  test("calculate angle @ 12:59") {
    calcAngle(12, 59) === 354
  }
}

File: src/test/java/com/cenerino/algorithms/java/misc/ClockHandsAngleTest.java
package com.cenerino.algorithms.java.misc;

import org.junit.Test;

import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.assertThat;

public class ClockHandsAngleTest {

    @Test
    public void shouldCalculateAngleAt00_00() {
        assertThat(ClockHandsAngle.calcAngle(0, 0), is(0));
    }

    @Test
    public void shouldCalculateAngleAt00_30() {
        assertThat(ClockHandsAngle.calcAngle(0, 30), is(180));
    }

    @Test
    public void shouldCalculateAngleAt03_45() {
        assertThat(ClockHandsAngle.calcAngle(3, 45), is(180));
    }

    @Test
    public void shouldCalculateAngleAt12_45() {
        assertThat(ClockHandsAngle.calcAngle(12, 45), is(270));
    }

    @Test
    public void shouldCalculateAngleAt15_00() {
        assertThat(ClockHandsAngle.calcAngle(15, 00), is(90));
    }

    @Test
    public void shouldCalculateAngleAt23_55() {
        assertThat(ClockHandsAngle.calcAngle(23, 55), is(0));
    }

    @Test
    public void shouldCalculateAngleAt23_59() {
        assertThat(ClockHandsAngle.calcAngle(23, 59), is(24));
    }

    @Test
    public void shouldCalculateAngleAt12_59() {
        assertThat(ClockHandsAngle.calcAngle(12, 59), is(354));
    }
}
